libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/LaneIconProcessor.kt

```diff
@@ -1,237 +1,310 @@
-                laneIndicator.directions.contains(STRAIGHT) &&
-                laneIndicator.directions.contains(RIGHT) -> {
-                activeDirection?.let {
-                    when {
-                        it.contains(LEFT, ignoreCase = true) -> {
-                            laneIcon[LaneTurns.LANE_LEFT_STRAIGHT_RIGHT_LEFT_ONLY]
-                        }
-                        it.contains(RIGHT, ignoreCase = true) -> {
-                            laneIcon[LaneTurns.LANE_LEFT_STRAIGHT_RIGHT_RIGHT_ONLY]
-                        }
-                        else -> {
-                            laneIcon[LaneTurns.LANE_LEFT_STRAIGHT_RIGHT_STRAIGHT_ONLY]
-                        }
-                    }
-                }
+            availableLaneNames.contains(laneName) -> {
+                LaneIcon(availableLaneNames[laneName]!!, shouldFlip)
             }
-            laneIndicator.directions.size > 2 &&
-                laneIndicator.directions.contains(SLIGHT_LEFT) &&
-                laneIndicator.directions.contains(STRAIGHT) &&
-                laneIndicator.directions.contains(SLIGHT_RIGHT) -> {
-                activeDirection?.let {
-                    when {
-                        it.contains(LEFT, ignoreCase = true) -> {
-                            laneIcon[
-                                LaneTurns.LANE_SLIGHT_LEFT_STRAIGHT_SLIGHT_RIGHT_SLIGHT_LEFT_ONLY
-                            ]
-                        }
-                        it.contains(RIGHT, ignoreCase = true) -> {
-                            laneIcon[
-                                LaneTurns.LANE_SLIGHT_LEFT_STRAIGHT_SLIGHT_RIGHT_SLIGHT_RIGHT_ONLY
-                            ]
-                        }
-                        else -> {
-                            laneIcon[LaneTurns.LANE_SLIGHT_LEFT_STRAIGHT_SLIGHT_RIGHT_STRAIGHT_ONLY]
-                        }
-                    }
-                }
+            laneIndication.activeDirection != null -> {
+                // Draw "lane_laneIndication.activeDirection_using_laneIndication.activeDirection"
+                val active = laneIndication.activeDirection.replace(RIGHT, TURN).replace(LEFT, TURN)
+                val name = "$LANE $active $USING $active"
+                LaneIcon(availableLaneNames[name]!!, shouldFlip)
             }
-            laneIndicator.directions.size > 1 &&
-                laneIndicator.directions.contains(LEFT) &&
-                laneIndicator.directions.contains(RIGHT) -> {
-                activeDirection?.let {
-                    if (it.contains(LEFT, ignoreCase = true)) {
-                        laneIcon[LaneTurns.LANE_LEFT_RIGHT_LEFT_ONLY]
-                    } else {
-                        laneIcon[LaneTurns.LANE_LEFT_RIGHT_RIGHT_ONLY]
-                    }
-                }
+            else -> {
+                // Draw "lane_straight"
+                LaneIcon(availableLaneNames["$LANE $STRAIGHT"]!!, shouldFlip)
```

Comment:
As discussed, I still think that it's better to fall back to one of the actually possible turns for this lane (or a subset) than just hardcode "straight".

Reply:
This is very subjective and customers might react to it differently. Look at this comment here > https://github.com/mapbox/navigation-ui-resources/pull/26
```
I decided not to add canvases for the following lane configurations, which were exceedingly rare:

opposite slight turn or straight or sharp turn and opposite sharp turn or straight or slight turn (93 occurrences)
opposite sharp turn or slight turn and opposite slight turn or sharp turn (30 occurrences)
straight or turn or sharp turn (18 occurrences)
uturn or opposite slight turn (15 occurrences)
opposite turn or straight or sharp turn (12 occurrences)
straight or slight turn or sharp turn (8 occurrences)
straight or slight turn or uturn (3 occurrences)
opposite sharp turn or straight or sharp turn (3 occurrences)
straight or sharp turn or uturn (no occurrences)
turn or sharp turn or uturn (no occurrences)
```

As you can see the only combinations for which you would see a `lane_straight` is for the combination above which are very extremely rare. Some customers might like that we just say we don't know(by drawing straight) or some might like we are showing a subset. I think it's difficult to choose the behavior here and since these combinations are so rare, I feel it's ok to say we don't know. Also, let's take a design opinion here

/cc @d-prukop

Reply:
I personally don't feel like showing straight indicates that we don't know. It might look like a much bigger mistake than at least drawing a single lane correctly.

If we'd like to be clear that we don't know, why not introduce a dedicated icon for it?

Reply:
We have thought of `?` but we would need design input.

Reply:
First of all, keep in mind that this fallback only occurs if there’s nothing to highlight. So whatever we show here is only to make clear that there is a lane, so that the user doesn’t miscount when figuring out which lane to get into. A question mark would probably be more distracting than helpful for this purpose.

The most robust fallback would indeed be the largest subset of lane indications that are supported in a single icon. However, it’s a question of whether all this effort is necessary for the 19 affected roads in https://github.com/mapbox/navigation-ui-resources/pull/26#issuecomment-924307448, many of which are probably mistagged anyways.

The next most robust fallback would be to show any of the lane’s indications arbitrarily. There’s a risk of showing a left turn arrow where a right turn arrow would be _slightly_ less confusing, but it’s almost moot, so if you find it straightforward enough to select an indication at random, go for it.

Reply:
I understand that this is rare, and personally, I would still prefer selecting one of the actually viable turns but it's not a hill I'm willing to die on. If iOS is planning to use `straight` as a fallback, let's do that here as well.